File: src/common/ts/src/message-broker/factories/consumer-factory.ts
import { IConsumer } from '../interfaces/iconsumer';
import { IConsumerConfiguration } from '../interfaces/iconsumer-configuration';
import { KafkaConsumer } from '../kafka/kafka-consumer';

export const ConsumerType = {
  KAFKA: 'Kafka',
};

// Helps with creation of a IConsumer base on
// the configuration sent
export class ConsumerFactory {
  static getConsumer(metadata: IConsumerConfiguration): IConsumer | undefined {
    if (metadata.type === ConsumerType.KAFKA) {
      return new KafkaConsumer(metadata.broker);
    }
    return undefined;
  }
}

File: src/common/py/casho/mylibrary/kafka/factories/consumer_factory.py
from ..interfaces import IConsumer, ConsumerConfiguration
from ..Kafka_implementation.kafka_consumer import KafkaConsumer

class ConsumerFactory:
    @staticmethod
    def get_consumer(metadata: ConsumerConfiguration) -> IConsumer or None:
        if metadata.type == "Kafka":
            return KafkaConsumer(metadata.broker,metadata.consumer_group)
        return None